crucible-llvm/src/Lang/Crucible/LLVM/Intrinsics/Libc.hs

```diff
@@ -823,3 +873,52 @@ llvmAssertRtnOverride =
        do let err = AssertFailureSimError "Call to __assert_rtn"
           liftIO $ assert sym (falsePred sym) err
   )
+
+llvmGetenv
+  :: (IsSymInterface sym, HasPtrWidth wptr, wptr ~ ArchWidth arch)
+  => LLVMOverride p sym arch
+        (EmptyCtx ::> LLVMPointerType wptr)
+        (LLVMPointerType wptr)
+llvmGetenv =
+  let nm = "getenv" in
+  LLVMOverride
+  ( L.Declare
+    { L.decRetType = L.PtrTo $ L.PrimType $ L.Integer 8
+    , L.decName    = L.Symbol nm
+    , L.decArgs    = [ L.PtrTo $ L.PrimType $ L.Integer 8
+                     ]
+    , L.decVarArgs = False
+    , L.decAttrs   = []
+    , L.decComdat  = mempty
+    }
+  )
+  (Empty :> PtrRepr)
+  PtrRepr
+  (\_ sym _args -> liftIO $ mkNullPointer sym PtrWidth)
```

Comment:
This seems like a way to introduce unsoundness... If we go ahead and make this available we should probably document that it is a feature of the simulator. 

Imagine:
```c
char *should_launch_missiles = getenv("DO_BAD_STUFF");
if (should_launch_missiles)
  launch_missiles();
normal_operation();
```
Crucible would say code never launches missiles.

Reply:
It's true, this is a bit of a hack we introduced to get through some initialization code.  A more complete solution would be to allow the C environment variables to be somehow specified...

Reply:
Perhaps we could have an override that checks for whichever variables are being accessed during initialization and returns `nullptr` for those but then doesn't work for others? It would still be unsound, but a little less so :skull:

Reply:
For now, I'm going to leave the dummy `getenv`.  I've opened ticked #302 to track the issue of a more realistic implementation for `getenv`, `putenv`, etc.